### DISCUSSION TITLE
Does changing the master password decouple the updated database from the old one?

### DISCUSSION THREAD

**Discussion by jrvidal** [keepassxreboot/keepassxc#8978] [Q&A] [answered]
Say I have a database with a not-so-great password and I'm concerned it's fallen into nefarious hands. I change my contained secrets and then I change the master password. Would an attacker that was successful in cracking the first master password have any leverage in accessing the updated database?

(My layman impression is that some encryption schemes work with an immutable, encrypted key in the file itself [that the password helps decrypting], in which case the answer would be yes :see_no_evil: ).

**Comment by droidmonkey** [comment, marked as answer]
Every time you save your database a new key is generated via a random salt value. So knowing the transformed key for one instance of the file provides no value to a subsequent instance. However, knowing the credentials that generate that transformed key will obviously work to decrypt the second one. In your case changing the database credentials effectively creates a brand new database that the previous one will provide no value towards decrypting.
